servlet/security-clientcert/src/test/java/org/javaee7/servlet/security/clientcert/SecureServletWithCommonNameTest.java

```diff
@@ -0,0 +1,286 @@
+        }
+        
+        // Only test TLS v1.2 for now
+        System.setProperty("jdk.tls.client.protocols", "TLSv1.2");
+        
+        //Enable flag to allow connection through common name onlys
+        List<String> cmd = new ArrayList<>();
+
+        cmd.add("set");
+        cmd.add("configs.config.server-config.security-service.auth-realm.certificate.property.common-name-as-principal-name=true");
+
+        CliCommands.payaraGlassFish(cmd);
+
+        // Add the client certificate that we just generated to the trust store of the server.
+        // That way the server will trust our certificate.
+        // Set the actual domain used with -Dpayara_domain=[domain name] 
+        addCertificateToContainerTrustStore(clientCertificate);
+        
+        System.out.println("Adding user for glassfish-remote");
+
+        return create(WebArchive.class)
+                .addClasses(SecureServletWithCommonName.class)
+                .addAsWebInfResource((new File(WEBAPP_SRC + "/WEB-INF", "web.xml")))
+                .addAsWebInfResource((new File(WEBAPP_SRC + "/WEB-INF", "glassfish-web.xml")));
+    }
+
+    @Before
+    public void setup() throws FileNotFoundException, IOException {
+        
+        System.out.println("\n*********** SETUP START ***************************");
+        
+        webClient = new WebClient();
+
+        // First get the HTTPS URL for which the server is listening
+        baseHttps = ServerOperations.toContainerHttps(base);
+        if (baseHttps == null) {
+            throw new IllegalStateException("No https URL could be created from " + base);
+        }
+        
+        // ### Ask the server for its certificate and add that to a new local trust store
+        
+        // Server -> client : the trust store certificates are used to validate the certificate sent
+        // by the server
+        
+        X509Certificate[] serverCertificateChain = getCertificateChainFromServer(baseHttps.getHost(), baseHttps.getPort());
+        
+        if (!isEmpty(serverCertificateChain)) {
+            
+            System.out.println("Obtained certificate from server. Storing it in client trust store");
+        
+            String trustStorePath = createTempJKSTrustStore(serverCertificateChain);
+    
+            System.out.println("Reading trust store from: " + trustStorePath);
+            
+            webClient.getOptions().setSSLTrustStore(new File(trustStorePath).toURI().toURL(), "changeit", "jks");
+            
+            // If the use.cnHost property is we try to extract the host from the server
+            // certificate and use exactly that host for our requests.
+            // This is needed if a server is listening to multiple host names, for instance
+            // localhost and example.com. If the certificate is for example.com, we can't
+            // localhost for the request, as that will not be accepted.
+            if (System.getProperty("use.cnHost") != null) {
```

Comment:
Perhaps this whole 'if' should be moved up to line 145. as the null check on baseHttps will capture a null when use.cnHost is set

Reply:
@arjantijms could do with your input on this as the original author of the method